builtin/credential/approle/path_role.go

```diff
@@ -521,8 +521,26 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 	return logical.ListResponse(listItems), nil
 }
 
-// setRoleEntry grabs a write lock and stores the options on an role into the storage.
-// Also creates a reverse index from the role's RoleID to the role itself.
+// validRoleConstraints tells if the role has at least one constraint enabled
+func validRoleConstraints(role *roleStorageEntry) (bool, error) {
```

Comment:
You could still check if the role is nil -- just return an error. Presumably this function shouldn't be getting called on a nil role anyways, so it seems fine to just error in that case (a nil role by definition does not have valid constraints, it seems to me).